### PR TITLE
Doc: convert some docs to markdown

### PR DESCRIPTION
**Pull request by andythenorth**
Almost entirely just formatting and links, except I fixed a couple of typos, missing punctuation.

### CODE REVIEW

**Comment by glx22** [comment]
Hmm would be better to rename `docs/HOWTO_compile_lang_files.txt` using git (like you did for the other files), so the history is kept

**Comment by glx22** [comment]
It's done in 2 commits to keep history.
This PR should not be squashed else history for `docs/HOWTO_compile_lang_files.md` is lost.

File: docs/HOWTO_compile_lang_files.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# How to compile lang files (OpenTTD and strgen)
+
+Last updated:    2009-06-30
+
+## 1.0) strgen usage
```
Comment (LordAro) [inline docs/HOWTO_compile_lang_files.md]: Get rid of numbers

File: docs/HOWTO_compile_lang_files.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+strgen takes as argument a txt file and translates it to a lng file, allowing
+it to be used inside OpenTTD. strgen needs the master language file
+`english.txt` to work. Below are some examples of strgen usage.
+
+## 1.1) Examples
+
+### Example 1
+
+If you are in the root of your working copy (git repository), you should type
+`strgen/strgen -s lang lang/english.txt`
```
Comment (LordAro) [inline docs/HOWTO_compile_lang_files.md]: inconsistent placement of strgen binary, should be `./strgen`, for consistency

File: docs/HOWTO_compile_lang_files.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+themselves without access to [translator.openttd.org](https://translator.openttd.org/).
+
+Please note that your compiled language file will only be compatible with the OpenTTD version
+you have downloaded `english.txt`, the master language file, for. While this is
+not always true, namely when changes in the code have not touched language
+files, your safest bet is to assume this 'limitation'.
+
+As a first step you need to compile strgen. This is as easy as typing
+`'make strgen'`. You can download the precompile strgen from:
+[http://www.openttd.org/download-strgen](http://www.openttd.org/download-strgen)
```
Comment (LordAro) [inline docs/HOWTO_compile_lang_files.md:17]: surprised that this is actually a thing.

**Review by LordAro — COMMENTED** [review]
seems ok

**Review by LordAro — APPROVED** [review]

**Pull request merged**